src/test/java/org/jenkinsci/plugins/pipeline/utility/steps/template/SimpleTemplateEngineStepTest.java

```diff
@@ -0,0 +1,193 @@
+                        "}", true));
+        WorkflowRun run = j.assertBuildStatus(Result.FAILURE, p.scheduleBuild2(0).get());
+        j.assertLogContains(Messages.SimpleTemplateEngineStepExecution_fileNotFound("doesnotexist.txt"), run);
+    }
+
+    @Test
+    public void missingBindingWithText() throws Exception {
+        WorkflowJob p = j.jenkins.createProject(WorkflowJob.class, "p");
+        p.setDefinition(new CpsFlowDefinition(
+                "node {\n" +
+                        "def s = simpleTemplateEngine text:'abc'\n"+
+                        "}", true));
+        WorkflowRun run = j.assertBuildStatus(Result.FAILURE, p.scheduleBuild2(0).get());
+        j.assertLogContains(Messages.SimpleTemplateEngineStepExecution_missingBindings("simpleTemplateEngine"), run);
+    }
+
+    @Test
+    public void missingBindingWithFile() throws Exception {
+        WorkflowJob p = j.jenkins.createProject(WorkflowJob.class, "p");
+        p.setDefinition(new CpsFlowDefinition(
+                "node {\n" +
+                        "def s = simpleTemplateEngine file:'doesnotexist.txt'\n"+
+                        "}", true));
+        WorkflowRun run = j.assertBuildStatus(Result.FAILURE, p.scheduleBuild2(0).get());
+        j.assertLogContains(Messages.SimpleTemplateEngineStepExecution_missingBindings("simpleTemplateEngine"), run);
+    }
+
+    @Test
+    public void noError() throws Exception {
+        WorkflowJob p = j.jenkins.createProject(WorkflowJob.class, "p");
+        p.setDefinition(new CpsFlowDefinition(
+                "node() {\n"+
+                        "def binding = [firstname : \"Grace\",]\n"+
+                        "String text = '''Dear <%= firstname %> '''\n"+
+                        "String result = simpleTemplateEngine text:text, bindings: binding\n"+
+                        "assert result == 'Dear Grace'\n"+
+                        "}", true));
+        WorkflowRun run = j.assertBuildStatusSuccess(p.scheduleBuild2(0));
+        j.assertLogContains("simpleTemplateEngine", run);
+    }
+
+    @Test
+    public void incorrectTemplate() throws Exception {
+        WorkflowJob p = j.jenkins.createProject(WorkflowJob.class, "p");
+        p.setDefinition(new CpsFlowDefinition(
+                "node() {\n"+
+                        "def binding = [firstname : \"Grace\",]\n"+
+                        "String text = '''Dear <%= wrong %> '''\n"+
+                        "String result = simpleTemplateEngine text:text, bindings: binding\n"+
+                        "}", true));
+        WorkflowRun run = j.assertBuildStatus(Result.FAILURE, p.scheduleBuild2(0).get());
+        j.assertLogContains("simpleTemplateEngine", run);
+        j.assertLogContains("hudson.remoting.ProxyException: groovy.lang.MissingPropertyException: No such property: wrong for class: SimpleTemplateScript", run);
+    }
+
+    // TODO: The next test needs input stimulus that only works when runInSandbox is true
```

Comment:
Typically this should be something that the sandbox prevents, e.g. have the template be `<% out.print(System.getProperties) %>` or `<% Jenkins.getInstance() %>`, and then assert that the build fails and check the log messages for a `RejectedAccessException` for the relevant method.